### PR TITLE
Update core-types exercises

### PR DESCRIPTION
**Pull request by VictoriaVasys**
* Update hash_test with symbol exercises
* Create nested_collections exercises

@AliSchlereth & @mikedao please check out these updates, let me know if there are too many, too few, not complex enough, etc. I'll work on new enums tomorrow.

### CODE REVIEW

File: core-types/hash_test.rb [hunk 1]
```diff
@@ -59,64 +112,171 @@ def test_member
...
+    expected_tv_characters = __
+    expected_shape_sides = __
+
+    assert_equal expected_new_restaurants, new_restaurants
+    assert_equal expected_name_lengths, name_lengths
+    assert_equal expected_planet_rings, planet_rings
+    assert_equal expected_saturn, saturn
+    assert_equal expected_tv_characters, tv_characters
+    assert_equal expected_shape_sides, shape_sides
+  end
```
Comment (AliSchlereth) [inline core-types/hash_test.rb]: This one is a bit difficult to read. Did you want to call the method on the hash itself for learning purposes? If not maybe set the hashes to variables at the top and then call the methods on those variables?  Also the indentation is inconsistent between lines 203/204 and 210/211/212.

File: core-types/hash_test.rb [hunk 2]
```diff
@@ -59,64 +112,171 @@ def test_member
...
+    expected_shape_sides = __
+    
+    assert_equal expected_new_restaurants, new_restaurants
+    assert_equal expected_name_lengths, name_lengths
+    assert_equal expected_name_lengths_2, name_lengths_2
+    assert_equal expected_planet_rings, planet_rings
+    assert_equal expected_saturn, saturn
+    assert_equal expected_tv_characters, tv_characters
+    assert_equal expected_shape_sides, shape_sides
+  end
```
Comment (AliSchlereth) [inline core-types/hash_test.rb]: Same feedback as for merge_with_symbols

File: core-types/nested_collections.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+class NestedCollectionsTest < Minitest::Test
+  
+  def test_2D_array
+    skip
+    numbers = [[1, 2], [9, 4, 89], 9]
+    
+    assert_equal __, numbers.first
+    assert_equal __, numbers[0][1]
+    assert_equal __, numbers[1][3]
```
Comment (AliSchlereth) [inline core-types/nested_collections.rb]: Just double checking that you meant to call an index position that isn't in the array.

File: core-types/nested_collections.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    
+    assert_equal ['yellow', ['blue', 'rose']], __
+    assert_equal ['yellow', 'blue'], __
+    assert_equal "Blackberry", __
+    assert_equal "white", __
+    assert_equal "rose", __
+  end
+  
+end
```
Comment (AliSchlereth) [inline core-types/nested_collections.rb]: Love this Nested Collections section!

File: core-types/hash_test.rb (hunk 1 above)
Comment (VictoriaVasys) [reply core-types/hash_test.rb]: I agree; I was trying to make it consistent with the one above, but wanted to be more explicit with naming, so it got a bit longer. I'll update with your suggestions.

File: core-types/nested_collections.rb (hunk 4 above)
Comment (VictoriaVasys) [reply core-types/nested_collections.rb]: ❤️ 🌷 🌿

File: core-types/nested_collections.rb (hunk 3 above)
Comment (VictoriaVasys) [reply core-types/nested_collections.rb]: Yes, just to give an example of what happens when you do; is that too confusing?

**Pull request merged**